// @seller-center/styles/tailwind.config.js
// Default config - https://github.com/tailwindlabs/tailwindcss/blob/v1/stubs/defaultConfig.stub.js
// const designToken = require("@walmart-web/sc-design-tokens/tokens");
const designToken = require("@seller-center/design-tokens");

module.exports = {
  future: {
    // removeDeprecatedGapUtilities: true,
    // purgeLayersByDefault: true,
  },
  purge: [],
  // plugins: [],
  important: true,
  theme: {
    colors: designToken.color,
    fontFamily: designToken.fontFamily,
    fontSize: designToken.fontSize,
    fontWeight: designToken.fontWeight,
    // Breakpoint go here
    screens: {},
    spacing: designToken.spacing,
    zIndex: designToken.zIndex,
    boxShadow: designToken.boxShadow,
    // width: {}
  },
  variants: {
    //  textColor: ['responsive', 'hover', 'focus']
    accessibility: [],
    alignContent: [],
    alignItems: [],
    alignSelf: [],
    appearance: [],
    backgroundAttachment: [],
    backgroundClip: [],
    backgroundColor: ["hover"],
    backgroundImage: [],
    gradientColorStops: [],
    backgroundOpacity: [],
    backgroundPosition: [],
    backgroundRepeat: [],
    backgroundSize: [],
    borderCollapse: [],
    borderColor: [],
    borderOpacity: [],
    borderRadius: [],
    borderStyle: [],
    borderWidth: [],
    boxShadow: [],
    boxSizing: [],
    container: [],
    cursor: [],
    display: [],
    divideColor: [],
    divideOpacity: [],
    divideStyle: [],
    divideWidth: [],
    fill: [],
    flex: [],
    flexDirection: [],
    flexGrow: [],
    flexShrink: [],
    flexWrap: [],
    float: [],
    clear: [],
    fontFamily: [],
    fontSize: [],
    fontSmoothing: [],
    fontVariantNumeric: [],
    fontStyle: [],
    fontWeight: [],
    height: [],
    inset: [],
    justifyContent: [],
    justifyItems: [],
    justifySelf: [],
    letterSpacing: [],
    lineHeight: [],
    listStylePosition: [],
    listStyleType: [],
    margin: [],
    maxHeight: [],
    maxWidth: [],
    minHeight: [],
    minWidth: [],
    objectFit: [],
    objectPosition: [],
    opacity: [],
    order: [],
    outline: [],
    overflow: [],
    overscrollBehavior: [],
    padding: [],
    placeContent: [],
    placeItems: [],
    placeSelf: [],
    placeholderColor: [],
    placeholderOpacity: [],
    pointerEvents: [],
    position: [],
    resize: [],
    space: [],
    stroke: [],
    strokeWidth: [],
    tableLayout: [],
    textAlign: [],
    textColor: ["hover"],
    textOpacity: [],
    textDecoration: [],
    textTransform: [],
    userSelect: [],
    verticalAlign: [],
    visibility: [],
    whitespace: [],
    width: [],
    wordBreak: [],
    zIndex: [],
    gap: [],
    gridAutoFlow: [],
    gridTemplateColumns: [],
    gridAutoColumns: [],
    gridColumn: [],
    gridColumnStart: [],
    gridColumnEnd: [],
    gridTemplateRows: [],
    gridAutoRows: [],
    gridRow: [],
    gridRowStart: [],
    gridRowEnd: [],
    transform: [],
    transformOrigin: [],
    scale: [],
    rotate: [],
    translate: [],
    skew: [],
    transitionProperty: [],
    transitionTimingFunction: [],
    transitionDuration: [],
    transitionDelay: [],
    animation: [],
  },
  // 113 keys
  corePlugins: {
    preflight: true,
    container: true,
    accessibility: false,
    alignContent: true,
    alignItems: true,
    alignSelf: true,
    animation: true,
    appearance: false,
    backgroundAttachment: false,
    backgroundClip: false,
    backgroundColor: true,
    backgroundImage: true,
    backgroundOpacity: true,
    backgroundPosition: true,
    backgroundRepeat: true,
    backgroundSize: false,
    borderCollapse: false,
    borderColor: true,
    borderOpacity: false,
    borderRadius: true,
    borderStyle: true,
    borderWidth: true,
    boxShadow: true,
    boxSizing: true,
    clear: true,
    cursor: true,
    display: true,
    divideColor: false,
    divideOpacity: false,
    divideStyle: false,
    divideWidth: false,
    fill: true,
    flex: true,
    flexDirection: true,
    flexGrow: true,
    flexShrink: true,
    flexWrap: true,
    float: false,
    fontFamily: true,
    fontSize: true,
    fontSmoothing: false,
    fontStyle: true,
    fontVariantNumeric: false,
    fontWeight: true,
    gap: true,
    gradientColorStops: false,
    gridAutoColumns: true,
    gridAutoFlow: true,
    gridAutoRows: true,
    gridColumn: true,
    gridColumnEnd: true,
    gridColumnStart: true,
    gridRow: true,
    gridRowEnd: true,
    gridRowStart: true,
    gridTemplateColumns: true,
    gridTemplateRows: true,
    height: true,
    inset: true,
    justifyContent: true,
    justifyItems: true,
    justifySelf: true,
    letterSpacing: false,
    lineHeight: true,
    listStylePosition: true,
    listStyleType: true,
    margin: true,
    maxHeight: true,
    maxWidth: true,
    minHeight: true,
    minWidth: true,
    objectFit: true,
    objectPosition: true,
    opacity: true,
    order: true,
    outline: false,
    overflow: true,
    overscrollBehavior: true,
    padding: true,
    placeContent: false,
    placeholderColor: false,
    placeholderOpacity: false,
    placeItems: true,
    placeSelf: true,
    pointerEvents: true,
    position: true,
    resize: true,
    rotate: false,
    scale: false,
    skew: false,
    space: true,
    stroke: true,
    strokeWidth: true,
    tableLayout: false,
    textAlign: true,
    textColor: true,
    textDecoration: true,
    textOpacity: false,
    textTransform: false,
    transform: false,
    transformOrigin: false,
    transitionDelay: false,
    transitionDuration: false,
    transitionProperty: false,
    transitionTimingFunction: false,
    translate: false,
    userSelect: true,
    verticalAlign: true,
    visibility: true,
    whitespace: true,
    width: true,
    wordBreak: true,
    zIndex: true,
  },
};
