// @seller-center/core-components/dist/esm/components/src/Spinner/Spinner.js
import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../../_virtual/_rollupPluginBabelHelpers.js';
import React from 'react';
import classNames from 'classnames';
import getNodeId from '../utility/getNodeId.js';
import style from './Spinner.module.scss.js';

var Spinner = props => {
  var {
    a11yLabel = "Loading…",
    className,
    color = "gray",
    size = "large",
    spinnerProps = {}
  } = props,
      rest = _objectWithoutProperties(props, ["a11yLabel", "className", "color", "size", "spinnerProps"]);

  var uniqueTitleId = getNodeId("svg_title");
  return /*#__PURE__*/React.createElement("span", _extends({
    className: classNames(style.container, size === "large" && style.large, size === "small" && style.small, className),
    role: "status"
  }, rest), /*#__PURE__*/React.createElement("svg", _extends({
    "aria-labelledby": uniqueTitleId,
    className: classNames(color === "gray" && style.gray, color === "white" && style.white),
    role: "img",
    viewBox: "0 0 48 48",
    xmlns: "http://www.w3.org/2000/svg"
  }, spinnerProps), /*#__PURE__*/React.createElement("title", {
    id: uniqueTitleId
  }, a11yLabel), /*#__PURE__*/React.createElement("defs", null, /*#__PURE__*/React.createElement("path", {
    d: "M24.7 3c-1.593 0-2.859.89-2.859 1.969l.985 10.992c.093.656.914 1.172 1.875 1.172.984 0 1.804-.516 1.922-1.172l.96-10.992C27.584 3.89 26.319 3 24.702 3z",
    id: "ld-spinner-pill"
  })), /*#__PURE__*/React.createElement("use", {
    className: style.pill,
    href: "#ld-spinner-pill"
  }), /*#__PURE__*/React.createElement("use", {
    className: classNames(style.pill, style.pill2),
    href: "#ld-spinner-pill"
  }), /*#__PURE__*/React.createElement("use", {
    className: classNames(style.pill, style.pill3),
    href: "#ld-spinner-pill"
  }), /*#__PURE__*/React.createElement("use", {
    className: classNames(style.pill, style.pill4),
    href: "#ld-spinner-pill"
  }), /*#__PURE__*/React.createElement("use", {
    className: classNames(style.pill, style.pill5),
    href: "#ld-spinner-pill"
  }), /*#__PURE__*/React.createElement("use", {
    className: classNames(style.pill, style.pill5),
    href: "#ld-spinner-pill"
  })));
};

export { Spinner };
