// @seller-center/core-components/dist/esm/components/src/LineClamp/LineClamp.js
import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../../_virtual/_rollupPluginBabelHelpers.js';
import { useState, useEffect, createElement, createRef } from 'react';
import classNames from 'classnames';
import styles from './LineClamp.module.scss.js';

var webkitLineClamp = import('webkit-line-clamp');
/*
  We used to calculate this dynamically for whatever text style happened to be present.
  But since it's always BogleWeb, we've hardcoded it.
*/

var BOGLE_BASELINE_RATIO = 0.21739130434782605;

var getRelativeLineHeight = node => {
  var style = window.getComputedStyle(node);
  return parseFloat(style.lineHeight) / parseFloat(style.fontSize);
};

var LineClamp = props => {
  var {
    className,
    lines = 2
  } = props,
      rest = _objectWithoutProperties(props, ["className", "lines"]);

  var ref = /*#__PURE__*/createRef();
  var [descenderPadding, setDescenderPadding] = useState(0);
  useEffect(() => {
    var span = document.createElement("span"); // test for support before pulling in polyfill

    /* istanbul ignore else: branch coverage fix */

    if (typeof span.style.webkitLineClamp !== "undefined") {
      return;
    }
    /* istanbul ignore next: ignore browser specific case */


    webkitLineClamp.then(({
      webkitLineClamp: lineClamp
    }) => {
      lineClamp(ref.current, lines);
    });
  });
  useEffect(() => {
    /* istanbul ignore if: branch coverage fix */
    if (!ref.current) {
      /* istanbul ignore next */
      return;
    }

    var lineHeight = getRelativeLineHeight(ref.current);
    setDescenderPadding(Math.max(0, 1 - lineHeight + BOGLE_BASELINE_RATIO) / 2);
  }, [ref]);
  return /*#__PURE__*/createElement("span", _extends({
    className: classNames(className, styles.lineClamp),
    ref: ref // eslint-disable-next-line react/forbid-dom-props
    ,
    style: {
      WebkitLineClamp: lines,
      paddingBottom: "".concat(descenderPadding, "em"),
      marginBottom: "-".concat(descenderPadding, "em")
    }
  }, rest));
};

export { BOGLE_BASELINE_RATIO, LineClamp };
