// @seller-center/core-components/dist/esm/components/src/Radio/Radio.js
import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../../_virtual/_rollupPluginBabelHelpers.js';
import { forwardRef, createElement } from 'react';
import classNames from 'classnames';
import getNodeId from '../utility/getNodeId.js';
import style from './Radio.module.scss.js';

var Radio = /*#__PURE__*/forwardRef((props, ref) => {
  var {
    checked = false,
    className,
    disabled = false,
    id: initialId,
    label,
    name,
    onChange,
    radioProps = {},
    value
  } = props,
      rest = _objectWithoutProperties(props, ["checked", "className", "disabled", "id", "label", "name", "onChange", "radioProps", "value"]);

  var {
    className: radioClassName
  } = radioProps,
      radioRest = _objectWithoutProperties(radioProps, ["className"]);

  var id = getNodeId("radio", initialId);
  return /*#__PURE__*/createElement("label", _extends({
    className: classNames(style.label, checked && style.checked, disabled && style.isDisabled, className),
    htmlFor: id
  }, rest), /*#__PURE__*/createElement("input", _extends({
    checked: checked,
    className: classNames(style.input, radioClassName),
    disabled: disabled,
    id: id,
    name: name,
    onChange: onChange,
    ref: ref,
    type: "radio"
  }, value && {
    value
  }, radioRest)), /*#__PURE__*/createElement("span", {
    className: style.dot
  }, /*#__PURE__*/createElement("span", {
    className: style.dotFill
  })), label);
});
Radio.displayName = "Radio";

export { Radio };
